hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DatanodeAdminMonitorImpl.java

```diff
@@ -488,6 +488,54 @@ private boolean checkContainersReplicatedOnNode(TrackedNode dn)
     return underReplicated == 0 && unclosed == 0;
   }
 
+  public Map<String, List<ContainerID>> containersReplicatedOnNode(DatanodeDetails dn)
```

Comment:
Let's rename this method to something more appropriate, like `getContainersPendingReplication`.

Reply:
Also if it doesn't turn out to be too complicated, let's refactor the common code between this method and the one above to another common method. That way we don't have to maintain the same logic in two places.

Reply:
I have refactored the common code, Could you please take a look at it now?

Reply:
I still see the method name as `containersReplicatedOnNode` in the latest commit... am I missing something?

Reply:
I have addressed the above in this pr: https://github.com/apache/ozone/pull/6293